app/tls_crypt_v2.ml

```diff
@@ -28,7 +28,7 @@ end = struct
     let len = Array.fold_left max 0 Mirage_crypto.Cipher_block.AES.CTR.key_sizes in
     Mirage_crypto.Cipher_block.AES.CTR.of_secret (Cstruct.sub cs 0 len)
 
-  let hmac cs = Cstruct.sub cs 64 64
```

Comment:
Are you sure about that? In `openvpn`, we have this `define`:
```c
#define MAX_HMAC_KEY_LENGTH 64
```

Reply:
Yes, it's described here: https://github.com/OpenVPN/openvpn/blob/master/doc/tls-crypt-v2.txt#L65 (256 bits = 32 bytes)

Besides, we have this `#define` even though we use only the first 32 bytes of the encryption key:
```C
#define MAX_CIPHER_KEY_LENGTH 64
```

(IMO it's a bit messy to have these keys that we only use half of)

Reply:
Another argument for is that it actually authenticates the wrapped key now that we check the tag and computed tag are equal (and not the opposite).

Reply:
Hmmmhmm, ok we must clarify that because we got so trouble when we read the `openvpn` code.

Reply:
Yes, this needs to be documented. It seems it's a thing throughout the openvpn code to only use the first N bytes of the keys where N depends on the cipher/hmac. See https://github.com/OpenVPN/openvpn/blob/master/src/openvpn/crypto_openssl.c#L1226-L1241

Reply:
I tried to document it here: https://git.robur.coop/robur/openvpn-spec/pulls/7